### ISSUE TITLE
Is this compatible with CIV3 ?

### ISSUE THREAD

**Issue by sahanDissanayake** [jenssegers/codeigniter-advanced-images#4]
Just trying to get this to work, But does not works @jenssegers 

I have attached a zip folder with the full application and you can see if you visit the index I mean the default_action

The images are not loaded

https://drive.google.com/open?id=0B6D3etNOHhPTWGx5bHdiN29kVzg&authuser=0

**Comment by Ihabafia** [comment]
What is the error your are getting with Image library?

_Ihab Abu Afia_

**Comment by sahanDissanayake** [comment]
There are no errors, thats the problem :cry: , its just the image does not show up.

**Comment by sahanDissanayake** [comment]
I get this error when I try to load an image straight away through the link

``` javascript
A PHP Error was encountered

Severity: Runtime Notice

Message: Only variables should be passed by reference

Filename: controllers/Media.php

Line Number: 36

Backtrace:

File: /Applications/XAMPP/xamppfiles/htdocs/Images/application/controllers/Media.php
Line: 36
Function: _error_handler

File: /Applications/XAMPP/xamppfiles/htdocs/Images/index.php
Line: 292
Function: require_once
```

**Comment by sahanDissanayake** [comment]
![screen shot 2015-06-13 at 5 59 45 pm](https://cloud.githubusercontent.com/assets/5580680/8143224/03ac8266-11f6-11e5-8ea3-1c6a9f69feee.png)

**Comment by Ihabafia** [comment]
You have:
          <img src="<?php echo image("media/street.jpg", "medium"); ?>" />
and it suppose to be:
          <img src="<?php echo image("/media/street.jpg", "medium"); ?>" />

_Ihab Abu Afia_

**Comment by Ihabafia** [comment]
also make sure media.php to be Media.php if you are using CI3

_Ihab Abu Afia_

**Comment by sahanDissanayake** [comment]
Nah no luck, When I do <  img src="<?php echo image("/media/street.jpg", "medium"); ?>" />

The image src is pointing out of the source folder, you know

My Codeigniter setup is in

localhost/Images/ 

When I do <   img src="<?php echo image("/media/street.jpg", "medium"); ?>" />

The image link is created at
localhost/media/street.....

**Comment by sahanDissanayake** [comment]
Changed the name but still the same error, nothing changed

**Comment by sahanDissanayake** [comment]
I think there is an issue in Media.php with reference variables. Im not a php ninja, so not sure

**Comment by Ihabafia** [comment]
I am not sure, I think it's not related to the media setup, it's the XAMPP
probably.

I will try to dig more and see.

_Ihab Abu Afia_

**Comment by sahanDissanayake** [comment]
Cheers, Appreciate your time :)

**Comment by Ihabafia** [comment]
I made few modifications (minors) and it's working perfectly.
Are you using Mac or PC

_Ihab Abu Afia_

**Comment by sahanDissanayake** [comment]
Hey, Im using Mac,

**Comment by sahanDissanayake** [comment]
Where can I see your modifications please ?

**Comment by sahanDissanayake** [comment]
@Ihabafia Where can I see your modifications please ?

**Comment by sahanDissanayake** [comment]
hey, Could you tell me how you fixed it please ? :+1:

**Comment by Ihabafia** [comment]
You can chech the code that i sent you and see what was you were making
wrong

**Comment by sahanDissanayake** [comment]
Sorry, I did not get any code :cry:

**Comment by templeman** [comment]
I can confirm that this is still working with Codeigniter 3.x, at least with the minimum configuration. Specifically, I'm using the library with CI 3.1.6.

However, I'm *not* using the optional image_helper so I can't speak to that bit working or not.

**Comment by iAdesanyaDaniel** [comment]
Yes. I confirm it still works with CI 3.1.8, and the image_helper works fine too.

> ![screen shot 2015-06-13 at 5 59 45 pm](https://cloud.githubusercontent.com/assets/5580680/8143224/03ac8266-11f6-11e5-8ea3-1c6a9f69feee.png)

_Originally posted by @sahanDissanayake in https://github.com/jenssegers/codeigniter-advanced-images/issues/4#issuecomment-111678794_

I had this same issue too while setting it up. The reason is because, the extended _MY_image_lib_ library was not properly loaded.

I keep all third-party package in the _third party_ folder, so I had to include the _package path_, before it started working fine.
